lib/todo_detail.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'providers.dart';
+
+class TodoDetail extends ConsumerWidget {
+  const TodoDetail({super.key});
```

Comment:
以下のようにすると確実に `index` を受け取れます。`main.dart` に書きますが、ルーティングを工夫する必要があります。
```suggestion
  const TodoDetail({
      required this.index,
      super.key,
  });
  
  final int index;
```